notebooks/hello-universe/Interpreting_CNNs/Interpreting_CNNs.ipynb

```diff
@@ -0,0 +1,803 @@
+    "heatmap = (pred_masks * masks).sum(axis=0)\n",
+    "heatmap = heatmap / N / p1"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a8a6e994-1360-45d8-a6a4-0f8e3bc9bb24",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot the results next to the original image\n",
+    "fig, axes = plt.subplots(1, 2, figsize=(14, 5))\n",
+    "axes[0].imshow(image)\n",
+    "axes[0].set_title(\"orig_img\")\n",
+    "i = axes[1].imshow(heatmap, cmap=\"jet\", alpha=0.8)\n",
+    "fig.colorbar(i)\n",
+    "axes[1].set_title(\"heat_map\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2dfc889d-efc0-4275-918f-26c86d7d8402",
+   "metadata": {},
+   "source": [
+    "**Image caption**: The above image contains two panels in a horizontal row. The first panel shows the original image of a merginig galaxy candidate, and the second panel shows a heat map of the RISE map."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "93bcaa55-1884-4665-8040-61e16ab35c90",
+   "metadata": {},
+   "source": [
+    "**When examining the selected image**, we see a similar ring around the galactic center. Note that this is not the case for all astronomical images, or even all images in this dataset. Try playing around with the selected image to generate different saliency maps. For images where the central feature is relatively small, RISE's occlusion-based methodology can be unreliable."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bcdc67cb-5a3d-4504-a7be-85992a535154",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## 4. Ablation Analysis\n",
+    "Saliency maps provide an intuitive visual understanding of our model's focus and can be useful for understanding the physical relevance of our classification scheme. In order to understand the efficacy of our model's architecture, we can instead use ablation analysis to determine the most important layers of our model. Ablation analysis works by rebuilding our model without a specified layer of interest and testing and comparing the performance of a partial model. This method allows us to determine which layers of the model are most important, and even if some layers are reducing the capabilities of our model.\n",
+    "\n",
+    "In the exercise below, we will build and train four mini-models on the same data set as before. As this is an educational notebook, we will limit the training time of each model to five epochs. Results with these models may vary considerably due to these training constraints, but we highly encourage you to try modifying this section of the notebook for different results (see [Exercises](#Exercises/Extensions))\n",
+    "\n",
+    "Note that performing an ablation analysis will require training multiple models with the same architecture. This can be quite power intensive on personal computers, so it is recommended that devices be plugged in before running the analysis.\n",
```

Comment:
```suggestion
    "Note that performing an ablation analysis will require training multiple models with the same architecture. This can be quite compute intensive, so if you are running this notebook locally it is recommended that your device be plugged in before running the analysis.\n",
```
I get that the point here is "beware running on your laptop", but I think it's still more the point that it's compute intensive which means you might need your laptop plugged in.